### PR TITLE
[Model] Add Unsupervised Bipartite Sage

### PR DESCRIPTION
**Pull request by husimplicity**
to #4 
1. Add Unsupervised Bipartite Sage
2. Optimise id2idx_v2

### CODE REVIEW

File: graphlearn_torch/python/utils/common.py [hunk 1]
```diff
@@ -41,8 +41,13 @@ def get_free_port(host: str = 'localhost') -> int:
 
 
 def id2idx_v2(gid, book):
```
Comment (baoleai) [inline graphlearn_torch/python/utils/common.py:43]: Duplicated with tensor.py/id2idx

File: examples/hetero/bipartite_sage_unsup.py [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+  target = torch.cat(targets, dim=0).numpy()
+
+  return roc_auc_score(target, pred)
+
+
+for epoch in range(1, 21):
+  loss = train()
+  test_auc = test(test_loader)
+
+  print(f'Epoch: {epoch:02d}, Loss: {loss:4f}, '
```
Comment (baoleai) [inline examples/hetero/bipartite_sage_unsup.py:277]: How do the model AUC and training performance compare to PyG's？

File: graphlearn_torch/python/utils/common.py (hunk 1 above)
Comment (husimplicity) [reply graphlearn_torch/python/utils/common.py]: fixed

File: examples/hetero/bipartite_sage_unsup.py (hunk 2 above)
Comment (husimplicity) [reply examples/hetero/bipartite_sage_unsup.py:277]: Performance of GLT on one V100 16G GPU 16G with the same default parameter setting is about 3.1x speedup compared to PyG's and AUC is close to PyG's.

**Review by baoleai — APPROVED** [review]

**Review by LiSu — APPROVED** [review]

**Pull request merged**